simtools/model/model_parameter.py

```diff
@@ -0,0 +1,561 @@
+    ):
+        self._logger = logging.getLogger(__name__)
+        self._extra_label = None
+        self.io_handler = io_handler.IOHandler()
+        self.db = db
+        if mongo_db_config is not None:
+            self.db = db_handler.DatabaseHandler(mongo_db_config=mongo_db_config)
+
+        self._parameters = {}
+        self._derived = None
+        self.site = names.validate_site_name(site) if site is not None else None
+        self.name = (
+            names.validate_telescope_name(telescope_model_name)
+            if telescope_model_name is not None
+            else None
+        )
+        self.label = label
+        self.model_version = names.validate_model_version_name(model_version)
+        self._config_file_directory = None
+        self._config_file_path = None
+
+        self._load_parameters_from_db()
+
+        self.simtel_config_writer = None
+        self._added_parameter_files = None
+        self._is_config_file_up_to_date = False
+        self._is_exported_model_files_up_to_date = False
+
+    def get_parameter(self, par_name):
+        """
+        Get an existing parameter of the model.
+
+        Parameters
+        ----------
+        par_name: str
+            Name of the parameter.
+
+        Returns
+        -------
+        Value of the parameter
+
+        Raises
+        ------
+        InvalidModelParameter
+            If par_name does not match any parameter in this model.
+        """
+        try:
+            return self._parameters[par_name]
+        except KeyError:
+            pass
+        try:
+            return self.derived[par_name]
+        except (KeyError, ValueError) as e:
+            msg = f"Parameter {par_name} was not found in the model"
+            self._logger.error(msg)
+            raise InvalidModelParameter(msg) from e
+
+    def get_parameter_value(self, par_name, parameter_dict=None):
+        """
+        Get the value of an existing parameter of the model.
+
+        Parameters
+        ----------
+        par_name: str
+            Name of the parameter.
+        parameter_dict: dict
+            Dictionary with complete DB entry for the given parameter
+            (including the 'value', 'units' fields).
+
+        Returns
+        -------
+        Value of the parameter.
+
+        Raises
+        ------
+        InvalidModelParameter
+            If par_name does not match any parameter in this model.
+        """
+
+        parameter_dict = parameter_dict if parameter_dict else self.get_parameter(par_name)
+        try:
+            return parameter_dict.get("value")
+        except KeyError as exc:
+            self._logger.error(f"Parameter {par_name} does not have a value")
+            raise exc
+
+    def get_parameter_value_with_unit(self, par_name):
+        """
+        Get the value of an existing parameter of the model as an Astropy Quantity with its unit.\
+        If no unit is provided in the model, the value is returned without a unit.
```

Comment:
Yes and no. You are right that the function name is a bit misleading for parameters without a unit. But the way we need to use it, dimensionless parameter should not be a Quantity.

Is this ok or do you suggest a renaming of the function? A good suggestion for a name?

Reply:
ok, if this is used only in one place and that is how we need it. Although, I would tend to get the value with `.value` in the code using this function instead of directly in the return statement. anyways, we can keep it like this.